### PR TITLE
Audit 8: Mitigate overflow

### PR DESCRIPTION
**Pull request by fmiguelgarcia**
### new features

- New error `Multisig::Error::<T>::TooManySigners`: It raises up when the number of signers is greater than pending number of signer in `remove_multisig_signers_via_creator` function.

### modified logic

- [Bug fix: Integer underflow in `pallets/multisig/src/lib.rs` via `remove_multisig_signers_via_creator`](https://github.com/PolymathNetwork/polymesh-audit/issues/8)

### CODE REVIEW

**Review by Centril — APPROVED** [review]

**Review by vkomenda — APPROVED** [review]

**Review by CJP10 — APPROVED** [review]

File: pallets/multisig/src/lib.rs [hunk 1]
```diff
@@ -488,10 +488,9 @@ decl_module! {
             ensure!(Self::is_changing_signers_allowed(&multisig), Error::<T>::ChangeNotAllowed);
             let signers_len: u64 = u64::try_from(signers.len()).unwrap_or_default();
 
-            // NB: the below check can be underflow but that doesn't matter
-            // because the checks in the next loop will fail in that case.
+            let pending_num_of_signers = <NumberOfSigners<T>>::get(&multisig).saturating_sub(signers_len);
```
Comment (maxsam4) [inline pallets/multisig/src/lib.rs:491]: Let's use `checked_sub` here.

**Review by maxsam4 — CHANGES_REQUESTED** [review]

File: pallets/multisig/src/lib.rs (hunk 1 above)
Comment (fmiguelgarcia) [reply pallets/multisig/src/lib.rs]: @maxsam4 I've also added a new error to cover that underflow.

File: pallets/multisig/src/lib.rs [hunk 2]
```diff
@@ -675,6 +676,8 @@ decl_error! {
         MultisigMissingIdentity,
         /// Scheduling of a proposal fails
         FailedToSchedule,
+        /// More signers than required.
+        TooMuchSigners,
```
Comment (Centril) [inline pallets/multisig/src/lib.rs:680]:
```suggestion
        TooManySigners,
```

File: pallets/multisig/src/lib.rs [hunk 3]
```diff
@@ -488,7 +488,8 @@ decl_module! {
             ensure!(Self::is_changing_signers_allowed(&multisig), Error::<T>::ChangeNotAllowed);
             let signers_len: u64 = u64::try_from(signers.len()).unwrap_or_default();
 
-            let pending_num_of_signers = <NumberOfSigners<T>>::get(&multisig).saturating_sub(signers_len);
+            let pending_num_of_signers = <NumberOfSigners<T>>::get(&multisig).checked_sub(signers_len)
+                .ok_or(Error::<T>::TooMuchSigners)?;
```
Comment (Centril) [inline pallets/multisig/src/lib.rs:492]:
```suggestion
                .ok_or(Error::<T>::TooManySigners)?;
```

**Review by Centril — CHANGES_REQUESTED** [review]

**Review by Centril — APPROVED** [review]

**Comment by Centril** [comment]
LGTM; please also add a changelog :)

**Review by maxsam4 — APPROVED** [review]

**Pull request merged**